### PR TITLE
{tools}[GCCcore/8.2.0] Flang v20190329

### PR DESCRIPTION
**Pull request by ocaisa**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by ocaisa** [comment]
Requires https://github.com/easybuilders/easybuild-easyblocks/pull/1729

File: easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+homepage = 'https://github.com/flang-compiler/'
+description = """Fortran compiler, based on LLVM."""
+
+toolchain = {'name': 'GCCcore', 'version': '8.2.0'}
+# Do not set optarch to True: it will cause the build to fail
+toolchainopts = {'optarch': False}
+
+sources = [
+    {
+        'download_filename': 'flang_%(version)s.tar.gz',
```
Comment (ocaisa) [inline easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb]: Yes, all the files have the same name in the different repos...what a great idea!

**Comment by ocaisa** [comment]
I tested this with some simple examples and it seems to work, I'd welcome input from others more experienced with Fortran.

**Comment by ocaisa** [comment]
Test report for logically similar easyconfig available at https://gist.github.com/ocaisa/3b2af4352aab588af9bd0875e88bb7a8

File: easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+builddependencies = [
+    ('CMake', '3.13.3'),
+    ('Python', '2.7.15'),
+    ('libxml2', '2.9.8'),
+]
+
+assertions = True
+skip_all_tests = True
+
+moduleclass = 'tools'
```
Comment (geimer) [inline easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb]: @ocaisa Are you sure?  Shouldn't this be `compiler`?

Comment (geimer) [reply easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb]: And btw: Isn't this the full suite of compilers, that is, `clang`, `clang++`, and `flang`?

File: easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb]: 🤦‍♂

File: easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+checksums = [
+    '3aa0b1d1f06c789d6b1d971b1d2b601908d87377f69b14c2b507e0b943864581',  # flang-llvm-20190329.src.tar.gz
+    '1ee73aa1eef4ef7f75c96b24bf02445440602064c0074891ca5344a63f1fe5b5',  # openmp-7.1.0.src.tar.xz
+    'b8c621da53829f8c53bad73125556fb1839c9056d713433b05741f7e445199f2',  # flang-flang_20190329.src.tar.gz
+    '7286ad2d9935916f1a9fad0893a506f704d2fa70d774ebe93c087ee58c5af1b4',  # flang-driver-flang_20190329.src.tar.gz
+]
+
+
+dependencies = [
+    # since Clang is a compiler, binutils is a runtime dependency too
```
Comment (boegel) [inline easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb]: `s/Clang/Flang`?

File: easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb (hunk 2 above)
Comment (ocaisa) [reply easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb]: > @ocaisa Are you sure? Shouldn't this be `compiler`?

Yes, sorry will change.

> And btw: Isn't this the full suite of compilers, that is, `clang`, `clang++`, and `flang`?

No, what is installed is just flang. The `clang` in the temporary LLVM install (which is an LLVM fork) is actually what is pointed to by the link `flang`, in the install step I copy over `flang` which breaks the link and makes a hard copy.

Comment (ocaisa) [reply easybuild/easyconfigs/f/Flang/Flang-20190329-GCCcore-8.2.0-CUDA-10.1.105.eb]: I only install what they list in their pre-compiled tarballs.

**Comment by ocaisa** [comment]
Updated test report with latest version of easyblock (and logically similar easyconfig) available at https://gist.github.com/ocaisa/3b2af4352aab588af9bd0875e88bb7a8

**Comment by boegel** [comment]
@ocaisa Do you think it's still worth pursuing this, or shall we close this (now ancient) PR?

**Comment by ocaisa** [comment]
We should close this PR but only after we include (the new) flang in the clang build

**Comment by boegel** [comment]
@ocaisa Let's not keep ancient PRs open as a TODO list, open an issue for it if you want a reminder for that somewhere?

(this is currently the oldest open PR, so I really want to get this out of the way somehow)

**Comment by boegel** [comment]
Closing this, it's ancient enough that this isn't of much use anymore...
And it can still serve as a reference for people who want to build a recent `Flang` version, even when it's closed.

**Pull request closed without merging**